MigrationScript/KeyvaultHelperFunctions.ps1

```diff
@@ -37,4 +37,30 @@ function Get-SecretValue($uri)
 	$parsed = $secretValue.value | ConvertFrom-Json
 
     return $parsed
+}
+
+<#
+.SYNOPSIS
+Function to get a delete and purge a secret from keyvault.
+
+.PARAMETER keyVaultName
+Key vault name.
+
+.PARAMETER secretKey
+Secret name to be retrieved from key vault. 
+
+.EXAMPLE
+DeleteAndPurgeSecretFromKeyVault -keyVaultName $keyVaultName -secretKey $name
+#>
+function DeleteAndPurgeSecretFromKeyVault([string]$keyVaultName, [string]$secretKey) {
+	try {
+		Add-KeyVaultRoleAssignment -keyVaultName $keyVaultName;
+		Remove-AzKeyVaultSecret -VaultName $keyVaultName -Name $secretKey -PassThru -Force
+		Start-Sleep -s 15
```

Comment:
Why is this sleep required ?

Reply:
After the secret is deleted, we have to wait before trying to purge the secret, hence a sleep of 15s is added.
This code will execute for only a small portion of providers which failed for the first time.